plugins/cpp_metrics/parser/include/cppmetricsparser/cppmetricsparser.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CC_PARSER_CPPMETRICSPARSER_H
+#define CC_PARSER_CPPMETRICSPARSER_H
+
+#include <parser/abstractparser.h>
+#include <parser/parsercontext.h>
+
+#include <model/cppastnodemetrics.h>
+#include <model/cppastnodemetrics-odb.hxx>
+
+#include <model/cppfunction.h>
+#include <model/cppfunction-odb.hxx>
+
+#include <util/parserutil.h>
+#include <util/threadpool.h>
+
+namespace cc
+{
+namespace parser
+{
+  
+class CppMetricsParser : public AbstractParser
+{
+public:
+  CppMetricsParser(ParserContext& ctx_);
+  virtual ~CppMetricsParser();
+  virtual bool cleanupDatabase() override;
+  virtual bool parse() override;
+
+private:
+  void functionParameters();
+
+  std::unordered_set<model::FileId> _fileIdCache;
```

Comment:
What is the purpose of this `_fileIdCache`? Nothing is ever inserted in it, only erased and looked up, but it will always be empty.